spec/unit/utils/crypto/symmetric/recrypt_spec.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+require_relative '../../../../../app/utils/crypto/symmetric/aes256iv'
+
+describe Crypto::Symmetric::Recrypt do
+  let(:alice) { users(:alice) }
+
+  it 'recrypts given team' do
+    expect(Crypto::EncryptionAlgorithm).to receive(:latest).and_return(:AES256)
+
+    team = Team::Shared.create(alice, name: 'Team 1')
+
+    expect(team.encryption_algorithm).to eq 'AES256IV'
+
+    private_key = alice.decrypt_private_key('password')
+    described_class.new(alice, team, private_key)
```

Comment:
it's not obvious that new also performs the recrypt.
so as explained further up ... make sure you call perform explicit